### DISCUSSION TITLE
Adding latest macro placers

### DISCUSSION THREAD

**Discussion by luarss** [The-OpenROAD-Project/OpenROAD-flow-scripts#1193] [General]
I saw there were a few issues/PRs floating around to integrate the latest macro placers (DREAMPlace, Circuit Training). May I know what are the plans with these? 

- [DREAMPlace](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/447)
- [Circuit Training](https://github.com/The-OpenROAD-Project/OpenROAD/issues/1563)
- [MacroPlacement](https://github.com/TILOS-AI-Institute/MacroPlacement) - this is an open-source replication of Circuit Training, preferably better to reproduce this in the OR framework. 
- [AutoDMP](https://github.com/NVlabs/AutoDMP)

**Comment by oharboe** [comment]
There is ongoing work to improve mpl2 and make it default: https://github.com/The-OpenROAD-Project/OpenROAD/pull/3512

In the cases that I have tried, mpl2 is superior to mpl.

**Comment by luarss** [comment]
https://github.com/The-OpenROAD-Project/OpenROAD/pull/3617
@oharboe I am writing some documentation for mpl2, do you have any example scripts handy that we can use?

**Reply by oharboe** [reply]
No, so far it has not worked on mock-array.

I think mock-array will be an interesting example, because it is using horizontal abutment. It would be impressive if mpl2 knows how to do that: vertical abutment cant work as the clock is constrained to the top edge of the Element macro.
